hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -334,15 +334,19 @@ private static AddressDefinition pickLoopbackAddress(String host, int defaultPor
     private AddressDefinition pickMatchingAddress(Collection<InterfaceDefinition> interfaces) throws SocketException {
         Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
         boolean preferIPv4Stack = preferIPv4Stack();
+        boolean matchInterfaceDefinition = interfaces != null && !interfaces.isEmpty();
```

Comment:
You can use `CollectionUtil.isNotEmpty(interfaces)` as well

Reply:
Thanks, I will update it.